crates/wasm-mutate/src/mutators/peephole/eggsy/lang.rs

```diff
@@ -878,9 +872,91 @@ impl Lang {
     }
 }
 
+// To match memory like nodes by its inmediates
+macro_rules! match_mem {
+    ($l:ident, $self: ident, $other: ident) => {
+        if let (
+            Lang::$l {
+                static_offset,
+                align,
+                mem,
+                ..
+            },
+            Lang::$l {
+                static_offset: static_offset2,
+                align: align2,
+                mem: mem2,
+                ..
+            },
+        ) = ($self, $other)
+        {
+            return ::std::mem::discriminant($self) == ::std::mem::discriminant($other)
+                && static_offset == static_offset2
+                && align == align2
+                && mem == mem2;
+        }
+    };
+}
+
 impl egg::Language for Lang {
     fn matches(&self, other: &Self) -> bool {
-        ::std::mem::discriminant(self) == ::std::mem::discriminant(other)
+        match_mem!(I32Load, self, other);
+        match_mem!(I64Load, self, other);
+        match_mem!(F32Load, self, other);
+        match_mem!(F64Load, self, other);
+        match_mem!(I32Load8S, self, other);
+        match_mem!(I32Load8U, self, other);
+        match_mem!(I32Load16S, self, other);
+        match_mem!(I32Load16U, self, other);
+        match_mem!(I64Load8S, self, other);
+        match_mem!(I64Load8U, self, other);
+        match_mem!(I64Load16S, self, other);
+        match_mem!(I64Load16U, self, other);
+        match_mem!(I64Load32S, self, other);
+        match_mem!(I64Load32U, self, other);
+        match_mem!(I32Store, self, other);
+        match_mem!(I64Store, self, other);
+        match_mem!(F32Store, self, other);
+        match_mem!(F64Store, self, other);
+        match_mem!(I32Store8, self, other);
+        match_mem!(I32Store16, self, other);
+        match_mem!(I64Store8, self, other);
+        match_mem!(I64Store16, self, other);
+        match_mem!(I64Store32, self, other);
+
+        match (self, other) {
+            (Lang::I32(v), Lang::I32(v2)) => {
+                ::std::mem::discriminant(self) == ::std::mem::discriminant(other) && v == v2
```

Comment:
These discriminant comparisons here and elsewhere will always be true given that we've already entered the match arm where `self` is a `Lang::I32(_)` and `other` is a `Lang::I32(_)` as well.

Same for most of the other cases below.

Reply:
Done :)